lib/src/rules/use_string_buffer.dart

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.use_string_buffer;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:linter/src/analyzer.dart';
+import 'package:linter/src/util/dart_type_utilities.dart';
+
+const _desc = r'Use string buffer to compose strings.';
+
+const _details = r'''
```

Comment:
Given that this is a rule motivated by performance concerns, it would be good to state that in the details and to provide some concrete numbers (the bad case is N times slower than the good case).